src/main/java/com/google/sps/managers/DatastoreManager.java

```diff
@@ -0,0 +1,334 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.managers;
+
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import java.util.ArrayList;
+import com.google.gson.reflect.TypeToken;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import com.google.sps.data.User;
+import com.google.sps.data.Stage;
+import com.google.sps.data.Game;
+import com.google.sps.data.Coordinates;
+import com.google.sps.data.Pair;
+import com.google.sps.data.Hint;
+import com.google.sps.data.SinglePlayerProgress;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+
+/** Servlet responsible for implementing the functions in the interface. **/
+public class DatastoreManager implements IDataManager {
+
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  Gson gson = new Gson();
+
+  /**
+  * Creates or Replaces an entity of a single User data in datastore
+  * @param user a User variable representing a single instance of a user.
```

Comment:
user a User variable representing a single instance of a User.
- Capital 'U'

Reply:
noted